### DISCUSSION TITLE
Tags - most efficient method

### DISCUSSION THREAD

**Discussion by jcolinger3** [monicahq/monica#6603] [Q&A]
I'm new to Monica.
I'm not understanding how to efficiently tag a bunch of contacts at the same time.
Seems like a multi select / checkbox on the Contact list and bulk modify would be the most intuitive, but I'm not sure what I'm doing and don't see any options for this.

Anybody have tips on Tags?

**Comment by elliotblackburn** [comment]
I recently ran into this as well when bulk importing 100+ contacts via vCard. I ended up doing it manually although it took a considerable amount of time (about an hour). A way to bulk tag would be fantastic. I'd also be forever grateful for a way to add multiple relationships at once. IE: add 4 people to a contact as siblings, rather than doing one at a time.

This is really just an on-boarding problem for me though, now I've done the work I can't see it being an issue going forward for me personally.

**Comment by Szeraax** [comment]
I used the API and wrote a couple of small functions to batch tag my contacts. If I shared my powershell code, would that be helpful to you?

**Reply by bilogic** [reply]
yes

**Reply by Szeraax** [reply]
I'll share this weekend. Feel free to follow up if I don't reply within 5 days.

**Reply by Szeraax** [reply]
Honestly, I don't know if this code still works or not. Heck, I don't know if I even finished building all of these functions. I just haven't used it in any recent memory. Here you go: https://gist.github.com/Szeraax/c2a79f7dc0bb16c794b477618c2f12b3

**Reply by bilogic** [reply]
thanks man!
